### PR TITLE
posix: optimize (p)writev by calling vinode_allocate_interval once

### PR DESCRIPTION
**Pull request by marcinslusarz**
This change is [Reviewable](https://reviewable.io/reviews/pmem/pmemfile/121)

### CODE REVIEW

**Comment by krzycz** [comment]
:lgtm:

---

Reviewed 2 of 2 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/pmem/pmemfile/121#-:-KijLOlyDBuCmZ1j-7_a:bnfp4nl)*

File: tests/posix/rw/rw.cpp [hunk 1]
```diff
@@ -1363,6 +1363,9 @@ TEST_F(rw, pread)
 	ASSERT_EQ(pmemfile_unlink(pfp, "/file1"), 0);
 }
 
+#define POISON_PATTERN 0x66
```
Comment (GBuella) [inline tests/posix/rw/rw.cpp]: `constexpr char` maybe

File: tests/posix/rw/rw.cpp [hunk 2]
```diff
@@ -1363,6 +1363,9 @@ TEST_F(rw, pread)
...
+#define FILL_PATTERN(i) ((char)(0xc0 + i))
```
Comment (GBuella) [inline tests/posix/rw/rw.cpp]: `constexpr char fill_pattern(char i)` maybe

File: tests/posix/rw/rw.cpp [hunk 3]
```diff
@@ -1407,28 +1410,28 @@ TEST_F(rw, readv)
 	PMEMfile *f = prepare_file(pfp);
 	ASSERT_NE(f, nullptr);
 
-	const int vec_size = 40;
+	const size_t vec_size = 40;
+	const size_t arr_len = 5;
```
Comment (GBuella) [inline tests/posix/rw/rw.cpp]: how about `constexpr size_t` ?

File: tests/posix/rw/rw.cpp (hunk 1 above)
Comment (marcinslusarz) [inline tests/posix/rw/rw.cpp]: Done.

File: tests/posix/rw/rw.cpp (hunk 2 above)
Comment (marcinslusarz) [inline tests/posix/rw/rw.cpp]: Done.

File: tests/posix/rw/rw.cpp (hunk 3 above)
Comment (marcinslusarz) [inline tests/posix/rw/rw.cpp]: Done.

**Pull request merged**